src/histogram/strategies.rs

```diff
@@ -143,11 +154,14 @@ impl<T> EquiSpaced<T>
     where
         T: Ord + Clone + FromPrimitive + NumOps + Zero
 {
-    /// **Panics** if `bin_width<=0`.
-    fn new(bin_width: T, min: T, max: T) -> Self
+    /// Returns `None` if `bin_width<=0` or `min` >= `max`.
+    fn new(bin_width: T, min: T, max: T) -> Option<Self>
```

Comment:
Question: is returning _no object_ really a valid outcome or should it actually return a `Result<Self, WhateverError>` to signify a failure?

Hi, curious/lurking reviewer here. Saw the first "adventure" blog post and ended up here. Really great work that you people are doing and an inspiring article/tutorial.

If returning `None` is genuinely a valid, non-exceptional outcome, given certain args, then maybe a function name other than `new` would help clarify it's behavior? Just a suggestion.

Reply:
A `Result` is indeed a better return type there - I've been lazy because `EquiSpaced` is not part of the public API, but that's not a reason to be sloppy.

The creation method for the strategy is called `from_array`: do you think an `Option` is a good output type or would rather see a `Result` as a user?

Reply:
I totally understand 👍 

As for the public `from_array` I'm hesitant to give a strong opinion yet, because I've only just started looking through the code. But the general rule of thumb I would apply is this:

* Use an `Option` if returning `None` means it's ok to continue
* If it's not ok to continue, the user may only see a problem later down the line, when a "symptomatic" error occurs rather than the actual cause.
* With `None` that symptom may just be no-output.

Is that any help?

Reply:
It totally is - going again through the code I think it makes sense to convert the output type to `Result`. I'll work on it :+1: 
Thanks for your input @munckymagik!

Reply:
Refactored to return `Result` - let me know what you think @munckymagik @jturner314